sympy/solvers/solveset.py

```diff
@@ -115,7 +115,17 @@ def invert_real(f_x, y, x, domain=S.Reals):
     Inverts a real-valued function. Same as _invert, but sets
     the domain to ``S.Reals`` before inverting.
     """
-    return _invert(f_x, y, x, domain)
+    lhs, rhs_s = _invert(f_x, y, x, domain)
+    if lhs == x and isinstance(rhs_s, FiniteSet):
+        for rhs in rhs_s:
+            try:
+                if checksol(f_x - y, x, rhs):
+                    pass
+                else:
+                    return (f_x, FiniteSet(y))
+            except:
```

Comment:
I think I should add `TypeError`, because that would only be where `checksol` won't be able to solve. Are there any other exceptions I need to include.